src/CommunityToolkit.Maui.Camera/Providers/CameraProvider.shared.cs

```diff
@@ -5,9 +5,50 @@
 /// </summary>
 partial class CameraProvider : ICameraProvider
 {
+	readonly Lock refreshLock = new();
+	Task? refreshTask;
+
+	internal partial ValueTask PlatformRefreshAvailableCameras(CancellationToken token);
+
 	/// <inheritdoc/>
 	public IReadOnlyList<CameraInfo>? AvailableCameras { get; private set; }
 
 	/// <inheritdoc/>
-	public partial ValueTask RefreshAvailableCameras(CancellationToken token);
-}
\ No newline at end of file
+	public bool IsInitialized => refreshTask?.IsCompletedSuccessfully is true;
+
+	bool IsNotRefreshing => refreshTask is null || refreshTask.IsCompleted;
+
+	/// <inheritdoc/>	
+	public ValueTask InitializeAsync(CancellationToken token)
+	{
+		lock (refreshLock)
+		{
+			if (IsInitialized)
+			{
+				return ValueTask.CompletedTask;
+			}
+
+			if (IsNotRefreshing)
+			{
+				refreshTask = PlatformRefreshAvailableCameras(token).AsTask();
+			}
+
+			return new ValueTask(refreshTask!);
```

Comment:
We don't accept the null forgiving operator so we will need to remove the ! 

I think other comments might affect this being modified anyway though